test/parallel/test-cli-eval.js

```diff
@@ -229,3 +229,48 @@ child.exec(`${nodejs} --use-strict -p process.execArgv`,
     assert.strictEqual(err, null);
   }));
 });
+
+// ESModule eval tests
+
+
+// Assert that "42\n" is written to stdout on module eval.
+const execOptions = '--experimental-modules --input-type module';
+child.exec(
+  `${nodejs} ${execOptions} --eval "console.log(42)"`,
+  common.mustCall((err, stdout) => {
+    assert.ifError(err);
+    assert.strictEqual(stdout, '42\n');
+  }));
+
+// Assert that "42\n" is written to stdout with print option.
+child.exec(
+  `${nodejs} ${execOptions} --print --eval "42"`,
+  common.mustCall((err, stdout) => {
+    assert.ifError(err);
+    assert.strictEqual(stdout, '42\n');
+  }));
+
+// Assert that error is written to stderr on invalid input.
+child.exec(
+  `${nodejs} ${execOptions} --eval "!!!!"`,
+  common.mustCall((err, stdout, stderr) => {
+    assert.ok(err);
+    assert.strictEqual(stdout, '');
+    assert.ok(stderr.indexOf('SyntaxError: Unexpected end of input') > 0);
+  }));
+
+// Assert that require is undefined in ESM support
+child.exec(
+  `${nodejs} ${execOptions} --eval "console.log(typeof require);"`,
+  common.mustCall((err, stdout) => {
+    assert.ifError(err);
+    assert.strictEqual(stdout, 'undefined\n');
+  }));
+
+// Assert that import.meta is defined in ESM
+child.exec(
+  `${nodejs} ${execOptions} --eval "console.log(typeof import.meta);"`,
```

Comment:
It could also be worth making this a test of `import.meta.url`. Note that such a test would be the exact test needed to fix https://github.com/nodejs/node/issues/28160#issuecomment-502471656.

Reply:
I would be a bit concerned that it would increase the test scope. E.g. this test is meant to verify that the source is run as a module. If we change the exact string in `import.meta.url` in the future, it would break this test even though the source is still running as a module. If we want to test specific behavior ("relative imports work in `--eval`"), I would prefer a dedicated test for that.